threads/timeminer.py

```diff
@@ -122,84 +116,36 @@ def ping_worker(self):
+                    continue
 
+                logger.info("(tick_worker) Pingpool not empty, building tick")
                 logger.debug("Tick stage--------------------------------------")
-                start = time.time()  # Start and end time used to adjust sleep
-
-                self.networker.stage = "tick"
 
                 self.generate_and_process_tick()
 
-                # All in all, there should be a total sleep of
-                # 'cycle_time_multiplier' * 'cycle_time' in this thread.
-                # Gets adjusted dynamically by wait_time mechanism above
-                end = time.time()
-
-                # Overshoot is used if we slept too long in ping stage,
-                # then we compensate in this tick stage by speeding up sleep
-                second_sleep = cycle_time - (end-start) - overshoot
-                second_sleep = 0 if second_sleep < 0 else second_sleep
-
-                time.sleep(second_sleep)  # 2nd sleep
-
-                logger.debug("Vote stage--------------------------------------")
-                start = time.time()
-
-                self.networker.stage = "vote"
-                # Use a ping to vote for highest continuity tick in tick_pool
-
-                # TODO: What happens if I just selfishly vote for my own tick?
-                active_tick_ref = self.clockchain.current_tick_ref()
-
-                self.generate_and_process_ping(active_tick_ref, vote=True)
-
-                logger.debug("Voted for: " + str(active_tick_ref))
-
-                end = time.time()
-
-                inbetween_sleep = cycle_time / 2
-                time.sleep(inbetween_sleep)  # 2.5th sleep
-
-                # Clearing ping_pool here already to possibly receive new pings
-                self.clockchain.ping_pool = {}
-
-                third_sleep = cycle_time - inbetween_sleep - (end-start)
-                third_sleep = 0 if third_sleep < 0 else third_sleep
-                time.sleep(third_sleep)  # 3rd sleep
-
-                logger.debug("Select ticks stage------------------------------")
-                self.networker.stage = "select"
-
-                self.clockchain.select_highest_voted_to_chain()
-                # TODO: If nothing was added to chain.. sth obv. wrong! Resync?
-
-                self.added_ping = False
-            else:
-                time.sleep(1)
+    def select_worker(self):
+        while True:
+            time.sleep(5)
+            if self.clockchain.tick_pool_size() > 0:
+                self.networker.stage = 'select'
+                logger.debug("Select stage--------------------------------------")
+                grace_period = 10
+                time.sleep(grace_period)
+
+                logger.debug("Number of ticks: " + str(self.clockchain.tick_pool_size()))
+                logger.debug("Number of forks: " + str(self.clockchain.fork_pool_size()))
+
+                if self.clockchain.fork_pool_size() > self.clockchain.tick_pool_size():
```

Comment:
reminder for problem when someone triggers your resync all the time

Reply:
Added a TODO